### PR TITLE
qt: add missing system_lib for QtNetwork

### PR DESCRIPTION
**Pull request by ericLemanissier**
closes conan-io/conan-center-index#5594
closes conan-io/conan-center-index#5566

Specify library name and version:  **qt/***

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

File: recipes/qt/6.x.x/conanfile.py [hunk 1]
```diff
@@ -785,6 +786,7 @@ def _create_plugin(pluginname, libname, type, requires):
                 self.cpp_info.components["qtCore"].system_libs.append("userenv")  # qtcore requires "__imp_GetUserProfileDirectoryW " which is in "UserEnv.Lib" library
                 self.cpp_info.components["qtCore"].system_libs.append("ws2_32")  # qtcore requires "WSAStartup " which is in "Ws2_32.Lib" library
                 self.cpp_info.components["qtNetwork"].system_libs.append("DnsApi")  # qtnetwork from qtbase requires "DnsFree" which is in "Dnsapi.lib" library
+                self.cpp_info.components["qtNetwork"].system_libs.append("Iphlpapi")
```
Comment (SpaceIm) [inline recipes/qt/6.x.x/conanfile.py:789]:
```suggestion
                self.cpp_info.components["qtNetwork"].system_libs.append("iphlpapi")
```

File: recipes/qt/6.x.x/conanfile.py [hunk 2]
```diff
@@ -785,6 +786,7 @@ def _create_plugin(pluginname, libname, type, requires):
                 self.cpp_info.components["qtCore"].system_libs.append("userenv")  # qtcore requires "__imp_GetUserProfileDirectoryW " which is in "UserEnv.Lib" library
                 self.cpp_info.components["qtCore"].system_libs.append("ws2_32")  # qtcore requires "WSAStartup " which is in "Ws2_32.Lib" library
                 self.cpp_info.components["qtNetwork"].system_libs.append("DnsApi")  # qtnetwork from qtbase requires "DnsFree" which is in "Dnsapi.lib" library
```
Comment (SpaceIm) [inline recipes/qt/6.x.x/conanfile.py:788]:
```suggestion
                self.cpp_info.components["qtNetwork"].system_libs.append("dnsapi")  # qtnetwork from qtbase requires "DnsFree" which is in "Dnsapi.lib" library
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]
lower case for mingw@linux

File: recipes/qt/6.x.x/conanfile.py [hunk 3]
```diff
@@ -775,7 +775,8 @@ def _create_plugin(pluginname, libname, type, requires):
             _create_module("WaylandClient", ["Gui", "wayland::wayland-client"])
             _create_module("WaylandCompositor", ["Gui", "wayland::wayland-server"])
 
-        self.cpp_info.components["qtCore"].cxxflags.append("-fPIC")
+        if self.settings.compiler != "Visual Studio":
```
Comment (SpaceIm) [inline recipes/qt/6.x.x/conanfile.py:778]:
```suggestion
        if self.settings.os != "Windows":
```

File: recipes/qt/6.x.x/conanfile.py [hunk 4]
```diff
@@ -775,7 +775,8 @@ def _create_plugin(pluginname, libname, type, requires):
...
+            self.cpp_info.components["qtCore"].cxxflags.append("-fPIC")
```
Comment (SpaceIm) [inline recipes/qt/6.x.x/conanfile.py:779]:
```suggestion
        if self.settings.os != "Windows":
            self.cpp_info.components["qtCore"].cxxflags.append("-fPIC")
```

**Review by SSE4 — APPROVED** [review]

File: recipes/qt/6.x.x/conanfile.py (hunk 4 above)
Comment (SpaceIm) [reply recipes/qt/6.x.x/conanfile.py:779]: by the way, why this flag in package_info()?

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/qt/6.x.x/conanfile.py (hunk 4 above)
Comment (ericLemanissier) [reply recipes/qt/6.x.x/conanfile.py]: this avoids falling into https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/global/qglobal.h#n1394

File: recipes/qt/6.x.x/conanfile.py [hunk 5]
```diff
@@ -775,7 +775,8 @@ def _create_plugin(pluginname, libname, type, requires):
             _create_module("WaylandClient", ["Gui", "wayland::wayland-client"])
             _create_module("WaylandCompositor", ["Gui", "wayland::wayland-server"])
 
-        self.cpp_info.components["qtCore"].cxxflags.append("-fPIC")
+        if self.settings.os != "Windows":
+            self.cpp_info.components["qtCore"].cxxflags.append("-fPIC")
```
Comment (AndreyMlashkin) [inline recipes/qt/6.x.x/conanfile.py:779]: Why do you add -fPic only to qtCore? What about all other modules?

Comment (ericLemanissier) [reply recipes/qt/6.x.x/conanfile.py:779]: all the other modules depend on qtCore, so they all get the flag for free

Comment (SpaceIm) [reply recipes/qt/6.x.x/conanfile.py:779]: Why is it ever added? It's an interface flag.

**Review by AndreyMlashkin — APPROVED** [review]

File: recipes/qt/6.x.x/conanfile.py (hunk 5 above)
Comment (ericLemanissier) [reply recipes/qt/6.x.x/conanfile.py:779]: this avoids falling into https://code.qt.io/cgit/qt/qtbase.git/tree/src/corelib/global/qglobal.h#n1394

**Review by uilianries — APPROVED** [review]

**Pull request merged**